src/components/AddBook.jsx

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { v4 as uuidv4 } from 'uuid';
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+
+function AddBook({ handleAdd }) {
+  const [text, setText] = useState('');
+  const [category, setCategory] = useState('');
+  const [btnDissable, setBtnDissable] = useState(true);
+  const [message, setMessage] = useState('');
+
+  const handleText = (e) => {
+    if (text === '') {
+      setBtnDissable(true);
+      setMessage(null);
+    } else {
+      setMessage(null);
+      setBtnDissable(false);
+    }
+    setText(e.target.value);
+  };
+
+  const handleCategory = (e) => {
+    if (text === 'Catigories') {
+      setBtnDissable(true);
+      setMessage('Please select category');
+    } else {
+      setMessage(null);
+      setBtnDissable(false);
+    }
+    setCategory(e.target.value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (text.trim().length > 2) {
+      const newTodo = {
+        id: uuidv4(),
+        category,
+        name: text,
+        director: 'Suzanne Collins',
+        completed: '0%',
+        chapter: 'Introduction',
+      };
+
+      handleAdd(newTodo);
+      setMessage(null);
+      setText('');
+    } else {
+      setMessage('Text must be at least 3 characters');
+      setTimeout(() => {
+        setMessage('');
+      }, 2000);
+    }
+  };
+
+  return (
+    <div className="py-5">
+      <h2 className="text-2xl text-gray-500 font-bold">Add new book</h2>
+      <form
+        className="py-5 flex gap-5 flex-col justify-between items-center md:flex-row :mdlex-wrap"
+        onSubmit={handleSubmit}
+      >
+        <div className="w-full">
+          <input
+            type="text"
+            onChange={handleText}
+            value={text}
+            placeholder="Book title"
+            className="py-2 px-4 w-full bg-white shadow-lg"
+          />
+        </div>
+        <div className="inline-block w-full md:w-7/12">
+          <select
+            onChange={handleCategory}
+            className="py-2 text-gray-400 bg-white w-full px-2 shadow-lg"
+          >
+            <option value="Catigories">Catigories</option>
+            <option value="Action">Action</option>
+            <option value="Science Fiction">Science Fiction</option>
+            <option value="Economy">Economy</option>
+          </select>
+        </div>
+        <button
+          type="submit"
+          disabled={btnDissable}
+          className="w-full md:w-5/12 bg-blue-500 py-2 text-white rounded-md hover:bg-blue-700 shadow-lg"
+        >
+          Add
+        </button>
+      </form>
```

Comment:
- [ ] [Optioanl] Well! the project requires a form with inputs for the title and author of the book, but your current form has inputs for the title and category. To align your project with the [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/project_init_project_wth_comps_v3_0.md#:~:text=IMPORTANT%20in%20the%20design%20you%20can%20see%20an%20input%20for%20a%20category%20%2D%20please%20replace%20it%20with%20an%20author.), please update your form to include inputs for both the title and author of the book, as specified. This will ensure that your project meets the intended objectives. Keep up the great work!